Source/Engine/Tools/MaterialGenerator/MaterialGenerator.Material.cpp

```diff
@@ -385,28 +385,76 @@ void MaterialGenerator::ProcessGroupMaterial(Box* box, Node* node, Value& value)
         // Tiling & Offset
     case 29:
     {
-        auto uv = tryGetValue(node->GetBox(0), getUVs).AsVector2();
-        auto tiling = tryGetValue(node->GetBox(1), node->Values[0]).AsVector2();
-        auto offset = tryGetValue(node->GetBox(2), node->Values[1]).AsVector2();
+        const auto uv = tryGetValue(node->GetBox(0), getUVs).AsVector2();
+        const auto tiling = tryGetValue(node->GetBox(1), node->Values[0]).AsVector2();
+        const auto offset = tryGetValue(node->GetBox(2), node->Values[1]).AsVector2();
 
         value = writeLocal(ValueType::Vector2, String::Format(TEXT("{0} * {1} + {2}"), uv.Value, tiling.Value, offset.Value), node);
         break;
     }
         // DDX
     case 30:
     {
-        auto inValue = tryGetValue(node->GetBox(0), 0, Value::Zero);
+        const auto inValue = tryGetValue(node->GetBox(0), 0, Value::Zero);
 
         value = writeLocal(inValue.Type, String::Format(TEXT("ddx({0})"), inValue.Value), node);
         break;
     }
         // DDY
     case 31:
     {
-        auto inValue = tryGetValue(node->GetBox(0), 0, Value::Zero);
+        const auto inValue = tryGetValue(node->GetBox(0), 0, Value::Zero);
 
         value = writeLocal(inValue.Type, String::Format(TEXT("ddy({0})"), inValue.Value), node);
         break;
+    }
+        // Sign
+    case 32:
+    {
+        const auto inValue = tryGetValue(node->GetBox(0), 0, Value::Zero);
+
+        value = writeLocal(ValueType::Float, String::Format(TEXT("sign({0})"), inValue.Value), node);
+        break;
+    }
+        // Any
+    case 33:
+    {
+        const auto inValue = tryGetValue(node->GetBox(0), 0, Value::Zero);
+
+        value = writeLocal(ValueType::Bool, String::Format(TEXT("any({0})"), inValue.Value), node);
+        break;
+    }
+        // All
+    case 34:
+    {
+        const auto inValue = tryGetValue(node->GetBox(0), 0, Value::Zero);
+
+        value = writeLocal(ValueType::Bool, String::Format(TEXT("all({0})"), inValue.Value), node);
+        break;
+    }
+        // Blackbody
+    case 35:
+    {
+        // Based on unity's implementation by using data gathered by Mitchell Charity.
```

Comment:
You mean this reference: http://www.vendian.org/mncharity/dir3/blackbody/ ?

Reply:
Yeah thats the one. I found an implementation on Unity's docs and it was the only good & working source that I could find regarding this node, other sources were either on websites that were completely outdated with broken links.

Reply:
Ok, could you replace that comment with sth like:
`// Reference: Mitchell Charity, http://www.vendian.org/mncharity/dir3/blackbody/`

Reply:
Done 👍🏻